userspace/libscap/engine/bpf/scap_bpf.c

```diff
@@ -706,7 +706,7 @@ static int32_t load_bpf_file(struct bpf_engine *handle) {
 			                      handle->m_filepath);
 		}
 
-		handle->elf = elf_begin(handle->program_fd, ELF_C_READ_MMAP_PRIVATE, NULL);
+		handle->elf = elf_begin(handle->program_fd, ELF_C_READ, NULL);
```

Comment:
Could you give a little more context on this one change? Is it needed because elftoolchain doesn't define `ELF_C_READ_MMAP_PRIVATE`? Do we know what the implication is? From naming alone it seems we would be allocating a block of memory and reading the file into it, rather than mmaping directly, am I correct?

Reply:
I don't think there's much of a difference. However, I've updated this to use either implementation depending on what's available.

Reply:
Cool, thanks!